src/v/kafka/server/handlers/txn_offset_commit.cc

```diff
@@ -101,13 +101,23 @@ ss::future<response_ptr> txn_offset_commit_handler::handle(
     } else if (!ctx.authorized(
                  security::acl_operation::write,
                  transactional_id{request.data.transactional_id})) {
-        return ctx.respond(txn_offset_commit_response{
-          request, error_code::transactional_id_authorization_failed});
+        auto ec = !ctx.audit()
+                    ? error_code::broker_not_available
+                    : error_code::transactional_id_authorization_failed;
+        return ctx.respond(txn_offset_commit_response{request, ec});
     } else if (!ctx.authorized(
                  security::acl_operation::read,
                  group_id{request.data.group_id})) {
+        auto ec = !ctx.audit() ? error_code::broker_not_available
+                               : error_code::group_authorization_failed;
+        return ctx.respond(txn_offset_commit_response{request, ec});
+    }
+
+    // TODO(oren): remove? or do we want an audit check after each round
+    // of authZ checks?
```

Comment:
Question for @michael-redpanda or @graphcareful

Reply:
see previous comment